src/wallet/wallet.cpp

```diff
@@ -1718,6 +1718,19 @@ CAmount CWallet::GetImmatureWatchOnlyBalance() const
     return nTotal;
 }
 
+CAmount CWallet::GetMinChange() const
+{
+    LOCK(cs_main);
+    // Coupling the minimum change with the minimum relay fee has two reasons:
+    // A user should be able to create transactions even though their wallet is
+    //  filled with small inputs: Small change avoidance would otherwise add
+    //  many unspents, such that the transaction exceeds the maximum allowed
+    //  size.
+    // However, creating change too small would instead fragment the user's
+    //  wallet into small unspents over time.
+    return ::minRelayTxFee.GetFee(MAX_STANDARD_TX_SIZE);
```

Comment:
I think that using MAX_STANDARD_TX_SIZE is way too large (it would be around 1 mBTC on my node).

The choice here is whether to add change, or instead just turn it into fee immediately. Turning it into change adds an output (34 bytes) and then later spending that change adds an input (144 bytes), so as long as the fee we would pay for those 178 bytes is less than the amount we would turn into change, it is worth it. You probably want a small constant factor (at least 3, like the dust rule itself uses) above that number, though,

You probably want a small multiple of that